components/client-attestation-mgt/org.wso2.carbon.identity.client.attestation.mgt/src/main/java/org/wso2/carbon/identity/client/attestation/mgt/validators/AppleAttestationValidator.java

```diff
@@ -0,0 +1,296 @@
+            return false;
+        }
+        byte[] authData = (byte[]) authDataObject;
+        byte[] rpIdHash = Arrays.copyOfRange(authData, 0, 32);
+
+        // Get the configured Apple App ID
+        String appID = clientAttestationMetaData.getAppleAppId();
+
+        if (StringUtils.isNotEmpty(appID)) {
+            // Calculate the SHA-256 hash of the App ID
+            byte[] appIdHash = getSHA256Hash(appID);
+
+            // Compare the rpIdHash with the calculated App ID hash
+            if (MessageDigest.isEqual(appIdHash, rpIdHash)) {
+                return true;
+            } else {
+                // If the comparison fails, update context and return false
+                clientAttestationContext.setAttested(false);
+                clientAttestationContext.setValidationFailureMessage("Attestation Auth data validation failed. " +
+                        "Replying party Id is not matched with app ID for application: " + applicationResourceId +
+                        " tenant: " + tenantDomain);
+                return false;
+            }
+        }
+        // If the App ID is not configured, update context and return false
+        clientAttestationContext.setAttested(false);
+        clientAttestationContext.setValidationFailureMessage("Attestation Auth data validation failed. " +
+                "App Id is not configured for application: " + applicationResourceId + " tenant: " + tenantDomain);
+        return false;
+    }
+
+    /**
+     * Computes the SHA-256 hash of the input string.
+     *
+     * @param input The input string for which the SHA-256 hash is to be computed.
+     * @return A byte array representing the SHA-256 hash of the input string.
+     * @throws ClientAttestationMgtException If the SHA-256 algorithm is not available.
+     */
+    private byte[] getSHA256Hash(String input) throws ClientAttestationMgtException {
+
+        try {
+            // Create a MessageDigest object using the SHA-256 algorithm
+            MessageDigest digest = MessageDigest.getInstance(SHA_256);
+
+            // Compute the hash by converting the input string to bytes and digesting it
+            return digest.digest(input.getBytes(StandardCharsets.UTF_8));
+        } catch (NoSuchAlgorithmException e) {
+            // Throw a client attestation exception if the SHA-256 algorithm is not available
+            throw new ClientAttestationMgtException("Unable to validate attestation, cause " +
+                    "SHA-256 algorithm is not available." , e);
+        }
+    }
+
+    private boolean isCertificateExpiringSoon(X509Certificate certificate) {
+
+        Date currentDate = new Date();
+        Date expirationDate = certificate.getNotAfter();
+
+        // Calculate the difference in days
+        long differenceInDays = (expirationDate.getTime() - currentDate.getTime()) / (24 * 60 * 60 * 1000);
+
+        // Check if the certificate is expiring within month.
+        return differenceInDays <= 30;
```

Comment:
Use the same constants that will be defined for https://github.com/wso2/carbon-identity-framework/pull/5165/files#r1393887838

Reply:
checked with team, 90 days is the best practise